File: Data Structures & Algorithms/pow-x-n/submission-1.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n == 1:
            return x
        elif n == - 1:
            return 1/x
        elif n == 0:
            return 1
        else:
            if n % 2 > 0:
                return self.myPow(x, int(n // 2)) * self.myPow(x, int(n // 2)) * self.myPow(x,1)
            else:
                return self.myPow(x, int(n // 2)) * self.myPow(x, int(n // 2))
    

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    List<Integer> lst = new ArrayList<>();
    public void dfs(TreeNode root)
    {
        if(root == null)
        {
            return;
        }
        dfs(root.left);
        lst.add(root.val);
        dfs(root.right);
    }
    public int kthSmallest(TreeNode root, int k) {
        dfs(root);
        return lst.get(k-1);
    }
}
